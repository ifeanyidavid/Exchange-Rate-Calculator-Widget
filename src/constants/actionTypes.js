export const FETCH_RATES_START = "FETCH_RATES_START";
export const FETCH_RATES_ERROR = "FETCH_RATES_ERROR";
export const FETCH_RATES_DONE = "FETCH_RATES_DONE";
export const UPDATE_SOURCE_CURRENCY = "UPDATE_SOURCE_CURRENCY";
export const UPDATE_DESTINATION_CURRENCY = "UPDATE_DESTINATION_CURRENCY";
export const EXCHANGE_CURRENCY = "EXCHANGE_CURRENCY";
