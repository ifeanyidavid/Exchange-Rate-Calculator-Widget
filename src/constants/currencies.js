export const currencies = {
  EUR: { id: 0, value: "EUR" },
  GBP: { id: 1, value: "GBP" },
  USD: { id: 2, value: "USD" }
};
