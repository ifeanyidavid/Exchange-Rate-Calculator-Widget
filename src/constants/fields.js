export const fieldNames = {
  sourceCurrency: "sourceCurrency",
  destinationCurrency: "destinationCurrency",
  sourceAmount: "sourceAmount",
  destinationAmount: "destinationAmount",
  balance: "balance",
  symbol: "symbol",
  description: "description",
  currency: "currency",
  id: "id",
  error: "error",
  loading: "loading",
  rates: "rates",
};
