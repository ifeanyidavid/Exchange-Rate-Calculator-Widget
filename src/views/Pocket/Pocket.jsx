import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import CurrencyPocketsCarousel from "../../components/CurrencyPocketsCarousel";
import ExchangeButton from "../../components/ExchangeButton";
import TransactionHistory from "../../components/TransactionHistory";

export const Wrapper = styled.div`
  background: rgb(48, 116, 234);
  background: -moz-linear-gradient(
    32deg,
    rgba(48, 116, 234, 1) 0%,
    rgba(48, 116, 234, 1) 70%,
    rgba(204, 0, 255, 1) 100%
  );
  background: -webkit-linear-gradient(
    32deg,
    rgba(48, 116, 234, 1) 0%,
    rgba(48, 116, 234, 1) 70%,
    rgba(204, 0, 255, 1) 100%
  );
  background: linear-gradient(
    32deg,
    rgba(48, 116, 234, 1) 0%,
    rgba(48, 116, 234, 1) 70%,
    rgba(204, 0, 255, 1) 100%
  );
  filter: progid:DXImageTransform.Microsoft.gradient(startColorstr="#3074ea",endColorstr="#cc00ff",GradientType=1);
  height: 450px;
  display: flex;
  flex-direction: column;
`;

function Pocket({
  setExchangeOpenState,
  updateSourceCurrency,
  pockets,
  sourceCurrency,
  transactionHistory
}) {
  return (
    <>
      <Wrapper>
        <CurrencyPocketsCarousel
          updateSourceCurrency={updateSourceCurrency}
          pockets={pockets}
          currency={sourceCurrency}
        />
        <ExchangeButton setExchangeOpenState={setExchangeOpenState} />
      </Wrapper>
      {transactionHistory.length > 0 && (
        <div className="p-3">
          <TransactionHistory transactionHistory={transactionHistory} />
        </div>
      )}
    </>
  );
}

export default Pocket;

Pocket.propTypes = {
  setExchangeOpenState: PropTypes.func.isRequired,
  updateSourceCurrency: PropTypes.func.isRequired,
  pockets: PropTypes.array.isRequired,
  sourceCurrency: PropTypes.string.isRequired,
  transactionHistory: PropTypes.array
};
