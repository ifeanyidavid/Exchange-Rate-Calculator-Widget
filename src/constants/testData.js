import { currencies } from "./currencies";
import { fieldNames } from "./fields";

export const testData = {
  rates: [
    {
      rates: { USD: 1.1075, GBP: 0.84195 },
      base: "EUR",
      date: "2019-12-09"
    },
    {
      rates: { EUR: 1.1877189857, USD: 1.3153987766 },
      base: "GBP",
      date: "2019-12-09"
    },
    {
      rates: { EUR: 0.9029345372, GBP: 0.7602257336 },
      base: "USD",
      date: "2019-12-09"
    }
  ],
  pocketsReducer: {
    transactionHistory: [
      {
        id: 0,
        description: `Exchanged to EUR`,
        time: new Date().toLocaleTimeString("en-GB"),
        sourceAmount: 23,
        destinationAmount: 42,
        sourceCurrency: "USD",
        destinationCurrency: "EUR"
      }
    ],
    pockets: {
      [currencies.EUR.value]: {
        [fieldNames.id]: currencies.EUR.id,
        [fieldNames.currency]: currencies.EUR.value,
        [fieldNames.balance]: 300,
        [fieldNames.symbol]: "€",
        [fieldNames.description]: "Euro"
      },
      [currencies.GBP.value]: {
        [fieldNames.id]: currencies.GBP.id,
        [fieldNames.currency]: currencies.GBP.value,
        [fieldNames.balance]: 300,
        [fieldNames.symbol]: "£",
        [fieldNames.description]: "British Pound"
      },
      [currencies.USD.value]: {
        [fieldNames.id]: currencies.USD.id,
        [fieldNames.currency]: currencies.USD.value,
        [fieldNames.balance]: 300,
        [fieldNames.symbol]: "$",
        [fieldNames.description]: "American Dollar"
      }
    }
  },
  sourceAmount: 23,
  destinationAmount: 54,
  sourceCurrency: "USD",
  destinationCurrency: "EUR",
  error: new Error("Network disconnected"),
  slideIndex: 1,
  pocketValues: [
    {
      [fieldNames.id]: currencies.EUR.id,
      [fieldNames.currency]: currencies.EUR.value,
      [fieldNames.balance]: 300,
      [fieldNames.symbol]: "€",
      [fieldNames.description]: "Euro"
    },
    {
      [fieldNames.id]: currencies.GBP.id,
      [fieldNames.currency]: currencies.GBP.value,
      [fieldNames.balance]: 300,
      [fieldNames.symbol]: "£",
      [fieldNames.description]: "British Pound"
    },
    {
      [fieldNames.id]: currencies.USD.id,
      [fieldNames.currency]: currencies.USD.value,
      [fieldNames.balance]: 300,
      [fieldNames.symbol]: "$",
      [fieldNames.description]: "American Dollar"
    }
  ]
};
