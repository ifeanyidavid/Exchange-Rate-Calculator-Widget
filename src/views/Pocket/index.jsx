import React from "react";
import Pocket from "./Pocket";

const PocketWrapper = props => <Pocket {...props} />;

export default PocketWrapper;
