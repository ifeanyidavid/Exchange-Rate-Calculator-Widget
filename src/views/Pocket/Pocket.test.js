import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import toJson from "enzyme-to-json";
import { currencies } from "../../constants/currencies";
import { fieldNames } from "../../constants/fields";
import Pocket from "../Pocket";
import { testData } from "../../constants/testData";

configure({ adapter: new Adapter() });

let [updateSourceCurrency, setExchangeOpenState] = new Array(2).fill(jest.fn());

function shallowSetup() {
  const pockets = testData.pocketValues;

  const transactionHistory = testData.pocketsReducer.transactionHistory;

  // sample props to pass to shallow render

  const props = {
    setExchangeOpenState: setExchangeOpenState,
    updateSourceCurrency: updateSourceCurrency,
    sourceCurrency: "EUR",
    pockets: pockets,
    transactionHistory: transactionHistory
  };

  // wrapper instance around rendered output
  const pocketEnzymeWrapper = shallow(<Pocket {...props} />);

  return { props, pocketEnzymeWrapper };
}

describe("Pockets component", () => {
  it("Renders correctly", () => {
    const { pocketEnzymeWrapper } = shallowSetup();
    expect(toJson(pocketEnzymeWrapper)).toMatchSnapshot();
  });
});
