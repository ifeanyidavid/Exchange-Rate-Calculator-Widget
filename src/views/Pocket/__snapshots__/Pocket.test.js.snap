// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pockets component Renders correctly 1`] = `
<Pocket
  pockets={
    Array [
      Object {
        "balance": 300,
        "currency": "EUR",
        "description": "Euro",
        "id": 0,
        "symbol": "€",
      },
      Object {
        "balance": 300,
        "currency": "GBP",
        "description": "British Pound",
        "id": 1,
        "symbol": "£",
      },
      Object {
        "balance": 300,
        "currency": "USD",
        "description": "American Dollar",
        "id": 2,
        "symbol": "$",
      },
    ]
  }
  setExchangeOpenState={[MockFunction]}
  sourceCurrency="EUR"
  transactionHistory={
    Array [
      Object {
        "description": "Exchanged to EUR",
        "destinationAmount": 42,
        "destinationCurrency": "EUR",
        "id": 0,
        "sourceAmount": 23,
        "sourceCurrency": "USD",
        "time": "2:35:40 PM",
      },
    ]
  }
  updateSourceCurrency={[MockFunction]}
/>
`;
